web/pdf_page_view.js

```diff
@@ -284,6 +288,34 @@ class PDFPageView {
     }
   }
 
+  #renderTextLayer() {
+    const { pdfPage, textLayer, viewport } = this;
+    if (!textLayer) {
+      return;
+    }
+
+    let error = null;
+    try {
+      if (!textLayer.renderingDone) {
+        const readableStream = pdfPage.streamTextContent({
+          includeMarkedContent: true,
+        });
+        textLayer.setTextContentStream(readableStream);
+      }
+      textLayer.render(viewport);
+    } catch (ex) {
+      console.error(`#renderTextLayer: "${ex}".`);
+      error = ex;
+    } finally {
+      this.eventBus.dispatch("textlayerrendered", {
```

Comment:
This will dispatch the event way too soon, before any textContent has either been fetched or rendered.
For this to work you'll need to make `TextLayerBuilder.render` return a Promise that only resolves one all parsing is actually done, and then `await textLayer.render(viewport);` above.